# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 6a0dcd8f4e3e
# Full Hash: 6a0dcd8f4e3e35d27edba6a4fef795cf8174ac78
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-02-18 04:16:06
# Regressor Bug: 1692350
# File Overlap Count: 1
# Description:
#   Bug 1692350 - Implement a telemetry probe to collect page load data for documents that use lazyload r=emilio,chutten
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104912
# ==============================================================================

diff -r 6d7590bfd8d3 -r 6a0dcd8f4e3e dom/base/Document.cpp
--- a/dom/base/Document.cpp	Wed Feb 17 17:08:51 2021 +0000
+++ b/dom/base/Document.cpp	Wed Feb 17 17:17:52 2021 +0000
@@ -15572,6 +15572,17 @@
   return wc && wc->ConsumeTransientUserGestureActivation();
 }
 
+void Document::IncLazyLoadImageCount() {
+  if (!mLazyLoadImageCount) {
+    if (WindowContext* wc = GetTopLevelWindowContext()) {
+      if (!wc->HadLazyLoadImage()) {
+        Unused << wc->SetHadLazyLoadImage(true);
+      }
+    }
+  }
+  ++mLazyLoadImageCount;
+}
+
 void Document::SetDocTreeHadMedia() {
   RefPtr<WindowContext> topWc = GetTopLevelWindowContext();
   if (topWc && !topWc->IsDiscarded() && !topWc->GetDocTreeHadMedia()) {